fastjsonschema/draft04.py

```diff
@@ -424,7 +424,7 @@ def generate_properties(self):
         with self.l('if {variable}_is_dict:'):
             self.create_variable_keys()
             for key, prop_definition in self._definition['properties'].items():
-                key_name = re.sub(r'($[^a-zA-Z]|[^a-zA-Z0-9])', '', key)
+                key_name = re.sub(r'($[^a-zA-Z]|[^a-zA-Z0-9])', '', key) if type(key) == str else key
```

Comment:
JSON only allows key names to be strings. If it's not a string, then it's JavaScript's object or Python's dict. Not sure if this should be allowed. If yes, it needs more work. Just on the next three lines is used key with quotation marks. It wouldn't work with numbers.

Reply:
I think you're right. Our previous json validation allowed this to exist by mistake, hence I thought the problem was here. I will close the PR.